models/room.js

```diff
@@ -217,10 +217,14 @@ RoomSchema.methods.look = function (socket, short) {
     output += `<span class="silver">${this.desc}</span>\n`;
   }
 
-  if (this.inventory && this.inventory.length > 0) {
+  if (this.inventory && this.inventory.filter(item => !item.hidden).length > 0) {
```

Comment:
`if(Array.isArray(this.inventory)` is a better way to make sure the item is an array, if you are doing that, you can skip the length check.